q2_diversity/_filter.py

```diff
@@ -12,13 +12,22 @@
 
 def filter_distance_matrix(distance_matrix: skbio.DistanceMatrix,
                            metadata: qiime2.Metadata,
-                           where: str=None) -> skbio.DistanceMatrix:
+                           where: str=None,
+                           exclude_ids: bool=False) -> skbio.DistanceMatrix:
     ids_to_keep = metadata.ids(where=where)
     # NOTE: there is no guaranteed ordering to output distance matrix because
     # `ids_to_keep` is a set, and `DistanceMatrix.filter` uses its iteration
     # order.
     try:
-        return distance_matrix.filter(ids_to_keep, strict=False)
+        filtered = distance_matrix.filter(ids_to_keep, strict=False)
```

Comment:
You're right, thank you.  And this also solves the problem which my elaborate and unnecessary code in the `except` was attempting to handle.